store/CSDK/CarbonRow.cpp

```diff
@@ -0,0 +1,129 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <jni.h>
+#include <cstring>
+#include "CarbonRow.h"
+
+CarbonRow::CarbonRow(JNIEnv *env) {
+    this->rowUtilClass = env->FindClass("org/apache/carbondata/sdk/file/RowUtil");
+    this->jniEnv = env;
+    getShortId = jniEnv->GetStaticMethodID(rowUtilClass, "getShort",
+        "([Ljava/lang/Object;I)S");
+    getIntId = jniEnv->GetStaticMethodID(rowUtilClass, "getInt",
+        "([Ljava/lang/Object;I)I");
+    getLongId = jniEnv->GetStaticMethodID(rowUtilClass, "getLong",
+        "([Ljava/lang/Object;I)J");
+    getDoubleId = jniEnv->GetStaticMethodID(rowUtilClass, "getDouble",
+        "([Ljava/lang/Object;I)D");
+    getFloatId = jniEnv->GetStaticMethodID(rowUtilClass, "getFloat",
+        "([Ljava/lang/Object;I)F");
+    getBooleanId = jniEnv->GetStaticMethodID(rowUtilClass, "getBoolean",
+        "([Ljava/lang/Object;I)Z");
+    getStringId = jniEnv->GetStaticMethodID(rowUtilClass, "getString",
+        "([Ljava/lang/Object;I)Ljava/lang/String;");
+    getDecimalId = jniEnv->GetStaticMethodID(rowUtilClass, "getDecimal",
+        "([Ljava/lang/Object;I)Ljava/lang/String;");
+    getVarcharId = jniEnv->GetStaticMethodID(rowUtilClass, "getVarchar",
+        "([Ljava/lang/Object;I)Ljava/lang/String;");
+    getArrayId = jniEnv->GetStaticMethodID(rowUtilClass, "getArray",
+        "([Ljava/lang/Object;I)[Ljava/lang/Object;");
+}
+
+void CarbonRow::setCarbonRow(jobject data) {
+    this->carbonRow = data;
+}
+
+short CarbonRow::getShort(int ordinal) {
+    jvalue args[2];
+    args[0].l = carbonRow;
+    args[1].i = ordinal;
+    return jniEnv->CallStaticShortMethodA(rowUtilClass, getShortId, args);
+}
+
+int CarbonRow::getInt(int ordinal) {
+    jvalue args[2];
+    args[0].l = carbonRow;
+    args[1].i = ordinal;
+    return jniEnv->CallStaticIntMethodA(rowUtilClass, getIntId, args);
+}
+
+long CarbonRow::getLong(int ordinal) {
+    jvalue args[2];
+    args[0].l = carbonRow;
+    args[1].i = ordinal;
+    return jniEnv->CallStaticLongMethodA(rowUtilClass, getLongId, args);
+}
+
+double CarbonRow::getDouble(int ordinal) {
+    jvalue args[2];
+    args[0].l = carbonRow;
+    args[1].i = ordinal;
+    return jniEnv->CallStaticDoubleMethodA(rowUtilClass, getDoubleId, args);
+}
+
+
+float CarbonRow::getFloat(int ordinal) {
+    jvalue args[2];
+    args[0].l = carbonRow;
+    args[1].i = ordinal;
+    return jniEnv->CallStaticFloatMethodA(rowUtilClass, getFloatId, args);
+}
+
+jboolean CarbonRow::getBoolean(int ordinal) {
+    jvalue args[2];
+    args[0].l = carbonRow;
+    args[1].i = ordinal;
+    return jniEnv->CallStaticBooleanMethodA(rowUtilClass, getBooleanId, args);
```

Comment:
(jboolean)carobnRow[ordinal]

just we can type cast like above ?

Reply:
can't, I tried before.